In [1]:
import pandas as pd
import numpy as np
import inflection
import plotly.express as px
import requests

In [2]:
# Set pandas to display all columns
pd.set_option('display.max_columns', None)

# Set pandas to display float numbers up to 2 decimal places
pd.set_option('display.float_format', lambda x: '%.2f' % x)

# Data exploration

In [3]:
df0 = pd.read_csv("dataset/zomato.csv")
df1 = df0.copy()
df1.head()

Restaurant ID             Restaurant Name  Country Code            City  \
0        6310675  Mama Lou's Italian Kitchen           162  Las Piñas City   
1        6310675  Mama Lou's Italian Kitchen           162  Las Piñas City   
2        6314542                   Blackbird           162     Makati City   
3        6301293                    Banapple           162     Makati City   
4        6315689                    Bad Bird           162     Makati City   

                                             Address  \
0  Block 1, Lot 36, Tropical Avenue Corner Tropic...   
1  Block 1, Lot 36, Tropical Avenue Corner Tropic...   
2  Nielson Tower, Ayala Triangle Gardens, Salcedo...   
3  Ayala Triangle Gardens, Salcedo Village, Makat...   
4  Hole In The Wall, Floor 4, Century City Mall, ...   

                                            Locality  \
0                                   BF International   
1                                   BF International   
2  Ayala Triangle Gardens, Salcedo Village, Makat...   
3  Ayala Triangle Gardens, Salcedo Village, Makat...   
4          Century City Mall, Poblacion, Makati City   

                                    Locality Verbose  Longitude  Latitude  \
0                   BF International, Las Piñas City     121.01     14.45   
1                   BF International, Las Piñas City     121.01     14.45   
2  Ayala Triangle Gardens, Salcedo Village, Makat...     121.02     14.56   
3  Ayala Triangle Gardens, Salcedo Village, Makat...     121.02     14.56   
4  Century City Mall, Poblacion, Makati City, Mak...     121.03     14.57   

                              Cuisines  Average Cost for two  \
0                              Italian                  1100   
1                              Italian                  1100   
2                      European, Asian                  3100   
3  Filipino, American, Italian, Bakery                   800   
4                             American                   700   

           Currency  Has Table booking  Has Online delivery  \
0  Botswana Pula(P)                  1                    0   
1  Botswana Pula(P)                  1                    0   
2  Botswana Pula(P)                  0                    0   
3  Botswana Pula(P)                  0                    0   
4  Botswana Pula(P)                  0                    0   

   Is delivering now  Switch to order menu  Price range  Aggregate rating  \
0                  0                     0            3              4.60   
1                  0                     0            3              4.60   
2                  0                     0            4              4.70   
3                  0                     0            3              4.40   
4                  0                     0            3              4.40   

  Rating color Rating text  Votes  
0       3F7E00   Excellent    619  
1       3F7E00   Excellent    619  
2       3F7E00   Excellent    469  
3       5BA829   Very Good    867  
4       5BA829   Very Good    858

In [4]:
print(f'O Dataset possui {df1.shape[0]} linhas e {df1.shape[1]} colunas.')

O Dataset possui 7527 linhas e 21 colunas.


In [5]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7527 entries, 0 to 7526
Data columns (total 21 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Restaurant ID         7527 non-null   int64  
 1   Restaurant Name       7527 non-null   object 
 2   Country Code          7527 non-null   int64  
 3   City                  7527 non-null   object 
 4   Address               7527 non-null   object 
 5   Locality              7527 non-null   object 
 6   Locality Verbose      7527 non-null   object 
 7   Longitude             7527 non-null   float64
 8   Latitude              7527 non-null   float64
 9   Cuisines              7512 non-null   object 
 10  Average Cost for two  7527 non-null   int64  
 11  Currency              7527 non-null   object 
 12  Has Table booking     7527 non-null   int64  
 13  Has Online delivery   7527 non-null   int64  
 14  Is delivering now     7527 non-null   int64  
 15  Switch to order menu 

In [6]:
df1.isnull().sum()

Restaurant ID            0
Restaurant Name          0
Country Code             0
City                     0
Address                  0
Locality                 0
Locality Verbose         0
Longitude                0
Latitude                 0
Cuisines                15
Average Cost for two     0
Currency                 0
Has Table booking        0
Has Online delivery      0
Is delivering now        0
Switch to order menu     0
Price range              0
Aggregate rating         0
Rating color             0
Rating text              0
Votes                    0
dtype: int64

In [7]:
df1.duplicated().sum()

585

In [8]:
for column in df1.columns:
    print(f'\033[1mColuna:\033[0m {column} - \033[1mValores únicos:\033[0m {len(df1[column].unique())} - \033[1mdType:\033[0m {df1[column].dtype}\n')
    print(df1[column].unique())
    print('\n ----------------')

Coluna: Restaurant ID - Valores únicos: 6942 - dType: int64

[6310675 6314542 6301293 ... 5923535 5914190 5913508]

 ----------------
Coluna: Restaurant Name - Valores únicos: 5914 - dType: object

["Mama Lou's Italian Kitchen" 'Blackbird' 'Banapple' ...
 'Boon Cafe & Restaurant' 'Kanaat Lokantası' 'Katibim']

 ----------------
Coluna: Country Code - Valores únicos: 15 - dType: int64

[162  30  14 216  37 184 214   1  94 148 215 166 189 191 208]

 ----------------
Coluna: City - Valores únicos: 125 - dType: object

['Las Piñas City' 'Makati City' 'Mandaluyong City' 'Manila'
 'Marikina City' 'Muntinlupa City' 'Pasay City' 'Pasig City' 'Quezon City'
 'San Juan City' 'Tagaytay City' 'Taguig City' 'Brasília' 'Rio de Janeiro'
 'São Paulo' 'Adelaide' 'Atlanta' 'Austin' 'Boston' 'Brisbane' 'Calgary'
 'Charlotte' 'Chicago' 'Dallas' 'Denver' 'Detroit' 'Houston' 'Las Vegas'
 'Los Angeles' 'Miami' 'Montreal' 'New Orleans' 'New York City' 'Ottawa'
 'Perth' 'Philadelphia' 'Phoenix' 'Portland' 'San 

In [9]:
df1.describe()

Restaurant ID  Country Code  Longitude  Latitude  Average Cost for two  \
count        7527.00       7527.00    7527.00   7527.00               7527.00   
mean     10556892.00         93.07      33.76     19.33               7152.11   
std       7075141.13         99.03      77.40     23.26             290606.83   
min           549.00          1.00    -122.70    -41.33                  0.00   
25%       3500060.00          1.00      -4.26     12.92                 60.00   
50%       7701457.00         30.00      73.79     25.25                290.00   
75%      17147150.50        214.00      79.83     31.64                600.00   
max      19040277.00        216.00     175.31     55.98           25000017.00   

       Has Table booking  Has Online delivery  Is delivering now  \
count            7527.00              7527.00            7527.00   
mean                0.06                 0.35               0.17   
std                 0.24                 0.48               0.38   
min                 0.00                 0.00               0.00   
25%                 0.00                 0.00               0.00   
50%                 0.00                 0.00               0.00   
75%                 0.00                 1.00               0.00   
max                 1.00                 1.00               1.00   

       Switch to order menu  Price range  Aggregate rating    Votes  
count               7527.00      7527.00           7527.00  7527.00  
mean                   0.00         2.62              4.12   616.40  
std                    0.00         0.96              0.68  1127.70  
min                    0.00         1.00              0.00     0.00  
25%                    0.00         2.00              4.00   152.00  
50%                    0.00         3.00              4.20   335.00  
75%                    0.00         3.00              4.50   663.00  
max                    0.00         4.00              4.90 41333.00

##### Erros encontrados

- Cuisines possui 15 linhas nulas (investigar)
- O dataset possui 585 linhas duplicadas
- Colunas com espaço nos nomes
- Switch to order menu possui apenas 1 valor em todas as linhas
- Possível outlier na coluna average cost for two
- Talvez converter a coluna `average cost for two` para dolar de acordo com a `currency`?


##### Sugestões CEO

- Transformar Country Code no nome dos países
- Transformar Código das cores no nome das cores
- Categorizar o price_range
- Utilizar apenas o primeiro tipo de "Cuisine" para restaurantes que possuam mais de um

# Data Cleaning

In [10]:
# Linhas Nulas

df1.dropna(inplace=True)

In [11]:
# Linhas duplicadas

df1.drop_duplicates(inplace=True)

In [12]:
# A coluna Switch to order menu não possui nenhuma informação

df1.drop('Switch to order menu', axis=1, inplace=True)

In [13]:
df1.shape

(6929, 20)

In [14]:
# Ajeitando o nome das colunas
df1.columns = list(map(lambda x: inflection.titleize(x), df1.columns))
df1.columns = list(map(lambda x: x.replace(" ", ""), df1.columns))
df1.columns = list(map(lambda x: inflection.underscore(x), df1.columns))

In [15]:
df1.columns

Index(['restaurant_id', 'restaurant_name', 'country_code', 'city', 'address',
       'locality', 'locality_verbose', 'longitude', 'latitude', 'cuisines',
       'average_cost_for_two', 'currency', 'has_table_booking',
       'has_online_delivery', 'is_delivering_now', 'price_range',
       'aggregate_rating', 'rating_color', 'rating_text', 'votes'],
      dtype='object')

In [16]:
# Transformando Country Code
countries = {
1: "India",
14: "Australia",
30: "Brazil",
37: "Canada",
94: "Indonesia",
148: "New Zeland",
162: "Philippines",
166: "Qatar",
184: "Singapure",
189: "South Africa",
191: "Sri Lanka",
208: "Turkey",
214: "United Arab Emirates",
215: "England",
216: "United States of America",
}

df1['country_code'] = df1['country_code'].apply(lambda x: countries[x])
df1.rename(columns={'country_code':'country'}, inplace=True)
df1.head()

restaurant_id             restaurant_name      country            city  \
0        6310675  Mama Lou's Italian Kitchen  Philippines  Las Piñas City   
2        6314542                   Blackbird  Philippines     Makati City   
3        6301293                    Banapple  Philippines     Makati City   
4        6315689                    Bad Bird  Philippines     Makati City   
5        6304833                       Manam  Philippines     Makati City   

                                             address  \
0  Block 1, Lot 36, Tropical Avenue Corner Tropic...   
2  Nielson Tower, Ayala Triangle Gardens, Salcedo...   
3  Ayala Triangle Gardens, Salcedo Village, Makat...   
4  Hole In The Wall, Floor 4, Century City Mall, ...   
5  Level 1, Greenbelt 2, Ayala Center, Greenbelt,...   

                                            locality  \
0                                   BF International   
2  Ayala Triangle Gardens, Salcedo Village, Makat...   
3  Ayala Triangle Gardens, Salcedo Village, Makat...   
4          Century City Mall, Poblacion, Makati City   
5              Greenbelt 2, San Lorenzo, Makati City   

                                    locality_verbose  longitude  latitude  \
0                   BF International, Las Piñas City     121.01     14.45   
2  Ayala Triangle Gardens, Salcedo Village, Makat...     121.02     14.56   
3  Ayala Triangle Gardens, Salcedo Village, Makat...     121.02     14.56   
4  Century City Mall, Poblacion, Makati City, Mak...     121.03     14.57   
5  Greenbelt 2, San Lorenzo, Makati City, Makati ...     121.02     14.55   

                              cuisines  average_cost_for_two  \
0                              Italian                  1100   
2                      European, Asian                  3100   
3  Filipino, American, Italian, Bakery                   800   
4                             American                   700   
5                             Filipino                   700   

           currency  has_table_booking  has_online_delivery  \
0  Botswana Pula(P)                  1                    0   
2  Botswana Pula(P)                  0                    0   
3  Botswana Pula(P)                  0                    0   
4  Botswana Pula(P)                  0                    0   
5  Botswana Pula(P)                  0                    0   

   is_delivering_now  price_range  aggregate_rating rating_color rating_text  \
0                  0            3              4.60       3F7E00   Excellent   
2                  0            4              4.70       3F7E00   Excellent   
3                  0            3              4.40       5BA829   Very Good   
4                  0            3              4.40       5BA829   Very Good   
5                  0            3              4.70       3F7E00   Excellent   

   votes  
0    619  
2    469  
3    867  
4    858  
5    930

In [17]:
df1.head()

restaurant_id             restaurant_name      country            city  \
0        6310675  Mama Lou's Italian Kitchen  Philippines  Las Piñas City   
2        6314542                   Blackbird  Philippines     Makati City   
3        6301293                    Banapple  Philippines     Makati City   
4        6315689                    Bad Bird  Philippines     Makati City   
5        6304833                       Manam  Philippines     Makati City   

                                             address  \
0  Block 1, Lot 36, Tropical Avenue Corner Tropic...   
2  Nielson Tower, Ayala Triangle Gardens, Salcedo...   
3  Ayala Triangle Gardens, Salcedo Village, Makat...   
4  Hole In The Wall, Floor 4, Century City Mall, ...   
5  Level 1, Greenbelt 2, Ayala Center, Greenbelt,...   

                                            locality  \
0                                   BF International   
2  Ayala Triangle Gardens, Salcedo Village, Makat...   
3  Ayala Triangle Gardens, Salcedo Village, Makat...   
4          Century City Mall, Poblacion, Makati City   
5              Greenbelt 2, San Lorenzo, Makati City   

                                    locality_verbose  longitude  latitude  \
0                   BF International, Las Piñas City     121.01     14.45   
2  Ayala Triangle Gardens, Salcedo Village, Makat...     121.02     14.56   
3  Ayala Triangle Gardens, Salcedo Village, Makat...     121.02     14.56   
4  Century City Mall, Poblacion, Makati City, Mak...     121.03     14.57   
5  Greenbelt 2, San Lorenzo, Makati City, Makati ...     121.02     14.55   

                              cuisines  average_cost_for_two  \
0                              Italian                  1100   
2                      European, Asian                  3100   
3  Filipino, American, Italian, Bakery                   800   
4                             American                   700   
5                             Filipino                   700   

           currency  has_table_booking  has_online_delivery  \
0  Botswana Pula(P)                  1                    0   
2  Botswana Pula(P)                  0                    0   
3  Botswana Pula(P)                  0                    0   
4  Botswana Pula(P)                  0                    0   
5  Botswana Pula(P)                  0                    0   

   is_delivering_now  price_range  aggregate_rating rating_color rating_text  \
0                  0            3              4.60       3F7E00   Excellent   
2                  0            4              4.70       3F7E00   Excellent   
3                  0            3              4.40       5BA829   Very Good   
4                  0            3              4.40       5BA829   Very Good   
5                  0            3              4.70       3F7E00   Excellent   

   votes  
0    619  
2    469  
3    867  
4    858  
5    930

In [18]:
# Criando a coluna com o nome das cores

colors = {
"3F7E00": "darkgreen",
"5BA829": "green",
"9ACD32": "lightgreen",
"CDD614": "orange",
"FFBA00": "red",
"CBCBC8": "darkred",
"FF7800": "darkred",
}

df1['color_nome'] = df1['rating_color'].apply(lambda x: colors[x])
df1.head()


restaurant_id             restaurant_name      country            city  \
0        6310675  Mama Lou's Italian Kitchen  Philippines  Las Piñas City   
2        6314542                   Blackbird  Philippines     Makati City   
3        6301293                    Banapple  Philippines     Makati City   
4        6315689                    Bad Bird  Philippines     Makati City   
5        6304833                       Manam  Philippines     Makati City   

                                             address  \
0  Block 1, Lot 36, Tropical Avenue Corner Tropic...   
2  Nielson Tower, Ayala Triangle Gardens, Salcedo...   
3  Ayala Triangle Gardens, Salcedo Village, Makat...   
4  Hole In The Wall, Floor 4, Century City Mall, ...   
5  Level 1, Greenbelt 2, Ayala Center, Greenbelt,...   

                                            locality  \
0                                   BF International   
2  Ayala Triangle Gardens, Salcedo Village, Makat...   
3  Ayala Triangle Gardens, Salcedo Village, Makat...   
4          Century City Mall, Poblacion, Makati City   
5              Greenbelt 2, San Lorenzo, Makati City   

                                    locality_verbose  longitude  latitude  \
0                   BF International, Las Piñas City     121.01     14.45   
2  Ayala Triangle Gardens, Salcedo Village, Makat...     121.02     14.56   
3  Ayala Triangle Gardens, Salcedo Village, Makat...     121.02     14.56   
4  Century City Mall, Poblacion, Makati City, Mak...     121.03     14.57   
5  Greenbelt 2, San Lorenzo, Makati City, Makati ...     121.02     14.55   

                              cuisines  average_cost_for_two  \
0                              Italian                  1100   
2                      European, Asian                  3100   
3  Filipino, American, Italian, Bakery                   800   
4                             American                   700   
5                             Filipino                   700   

           currency  has_table_booking  has_online_delivery  \
0  Botswana Pula(P)                  1                    0   
2  Botswana Pula(P)                  0                    0   
3  Botswana Pula(P)                  0                    0   
4  Botswana Pula(P)                  0                    0   
5  Botswana Pula(P)                  0                    0   

   is_delivering_now  price_range  aggregate_rating rating_color rating_text  \
0                  0            3              4.60       3F7E00   Excellent   
2                  0            4              4.70       3F7E00   Excellent   
3                  0            3              4.40       5BA829   Very Good   
4                  0            3              4.40       5BA829   Very Good   
5                  0            3              4.70       3F7E00   Excellent   

   votes color_nome  
0    619  darkgreen  
2    469  darkgreen  
3    867      green  
4    858      green  
5    930  darkgreen

In [19]:
# Categorizando o price range

price = {1:'cheap',
         2:'normal',
         3:'expensive',
         4:'gourmet'}

df1['price_range'] = df1['price_range'].apply(lambda x: price[x])
df1.head()

restaurant_id             restaurant_name      country            city  \
0        6310675  Mama Lou's Italian Kitchen  Philippines  Las Piñas City   
2        6314542                   Blackbird  Philippines     Makati City   
3        6301293                    Banapple  Philippines     Makati City   
4        6315689                    Bad Bird  Philippines     Makati City   
5        6304833                       Manam  Philippines     Makati City   

                                             address  \
0  Block 1, Lot 36, Tropical Avenue Corner Tropic...   
2  Nielson Tower, Ayala Triangle Gardens, Salcedo...   
3  Ayala Triangle Gardens, Salcedo Village, Makat...   
4  Hole In The Wall, Floor 4, Century City Mall, ...   
5  Level 1, Greenbelt 2, Ayala Center, Greenbelt,...   

                                            locality  \
0                                   BF International   
2  Ayala Triangle Gardens, Salcedo Village, Makat...   
3  Ayala Triangle Gardens, Salcedo Village, Makat...   
4          Century City Mall, Poblacion, Makati City   
5              Greenbelt 2, San Lorenzo, Makati City   

                                    locality_verbose  longitude  latitude  \
0                   BF International, Las Piñas City     121.01     14.45   
2  Ayala Triangle Gardens, Salcedo Village, Makat...     121.02     14.56   
3  Ayala Triangle Gardens, Salcedo Village, Makat...     121.02     14.56   
4  Century City Mall, Poblacion, Makati City, Mak...     121.03     14.57   
5  Greenbelt 2, San Lorenzo, Makati City, Makati ...     121.02     14.55   

                              cuisines  average_cost_for_two  \
0                              Italian                  1100   
2                      European, Asian                  3100   
3  Filipino, American, Italian, Bakery                   800   
4                             American                   700   
5                             Filipino                   700   

           currency  has_table_booking  has_online_delivery  \
0  Botswana Pula(P)                  1                    0   
2  Botswana Pula(P)                  0                    0   
3  Botswana Pula(P)                  0                    0   
4  Botswana Pula(P)                  0                    0   
5  Botswana Pula(P)                  0                    0   

   is_delivering_now price_range  aggregate_rating rating_color rating_text  \
0                  0   expensive              4.60       3F7E00   Excellent   
2                  0     gourmet              4.70       3F7E00   Excellent   
3                  0   expensive              4.40       5BA829   Very Good   
4                  0   expensive              4.40       5BA829   Very Good   
5                  0   expensive              4.70       3F7E00   Excellent   

   votes color_nome  
0    619  darkgreen  
2    469  darkgreen  
3    867      green  
4    858      green  
5    930  darkgreen

In [20]:
# Mantendo apenas um tipo de "cuisine"

df1['cuisines'] = df1['cuisines'].apply(lambda x: x.split(',')[0])
df1.head()

restaurant_id             restaurant_name      country            city  \
0        6310675  Mama Lou's Italian Kitchen  Philippines  Las Piñas City   
2        6314542                   Blackbird  Philippines     Makati City   
3        6301293                    Banapple  Philippines     Makati City   
4        6315689                    Bad Bird  Philippines     Makati City   
5        6304833                       Manam  Philippines     Makati City   

                                             address  \
0  Block 1, Lot 36, Tropical Avenue Corner Tropic...   
2  Nielson Tower, Ayala Triangle Gardens, Salcedo...   
3  Ayala Triangle Gardens, Salcedo Village, Makat...   
4  Hole In The Wall, Floor 4, Century City Mall, ...   
5  Level 1, Greenbelt 2, Ayala Center, Greenbelt,...   

                                            locality  \
0                                   BF International   
2  Ayala Triangle Gardens, Salcedo Village, Makat...   
3  Ayala Triangle Gardens, Salcedo Village, Makat...   
4          Century City Mall, Poblacion, Makati City   
5              Greenbelt 2, San Lorenzo, Makati City   

                                    locality_verbose  longitude  latitude  \
0                   BF International, Las Piñas City     121.01     14.45   
2  Ayala Triangle Gardens, Salcedo Village, Makat...     121.02     14.56   
3  Ayala Triangle Gardens, Salcedo Village, Makat...     121.02     14.56   
4  Century City Mall, Poblacion, Makati City, Mak...     121.03     14.57   
5  Greenbelt 2, San Lorenzo, Makati City, Makati ...     121.02     14.55   

   cuisines  average_cost_for_two          currency  has_table_booking  \
0   Italian                  1100  Botswana Pula(P)                  1   
2  European                  3100  Botswana Pula(P)                  0   
3  Filipino                   800  Botswana Pula(P)                  0   
4  American                   700  Botswana Pula(P)                  0   
5  Filipino                   700  Botswana Pula(P)                  0   

   has_online_delivery  is_delivering_now price_range  aggregate_rating  \
0                    0                  0   expensive              4.60   
2                    0                  0     gourmet              4.70   
3                    0                  0   expensive              4.40   
4                    0                  0   expensive              4.40   
5                    0                  0   expensive              4.70   

  rating_color rating_text  votes color_nome  
0       3F7E00   Excellent    619  darkgreen  
2       3F7E00   Excellent    469  darkgreen  
3       5BA829   Very Good    867      green  
4       5BA829   Very Good    858      green  
5       3F7E00   Excellent    930  darkgreen

In [21]:
# Criando uma nova coluna para receber o custo médio para duas pessoas convertido para USD

df1.currency.unique()

array(['Botswana Pula(P)', 'Brazilian Real(R$)', 'Dollar($)',
       'Emirati Diram(AED)', 'Indian Rupees(Rs.)',
       'Indonesian Rupiah(IDR)', 'NewZealand($)', 'Pounds(£)',
       'Qatari Rial(QR)', 'Rand(R)', 'Sri Lankan Rupee(LKR)',
       'Turkish Lira(TL)'], dtype=object)

In [22]:
def get_rates(to_currency='USD'):
    # Acessando a API para obter a taxa de conversão do dolar para todas as moedas
    url = 'https://api.exchangerate-api.com/v4/latest/'+str(to_currency)
    response = requests.get(url)
    data = response.json()

    return data

In [23]:
data = get_rates()
def convert_currency(amount, from_currency):
    # Salvando a taxa de conversão a partir do json recebido pela API
    exchange_rate = data['rates'][from_currency]
    # Convertendo o valor - Como a taxa que obtemos pela API é A PATIR do dolar, aqui vamos fazer a operação inversa (divisão) para obter a taxa de conversão PARA o dolar
    # Como a coluna original possuia apenas números inteiros, vamos arredondar para manter também números inteiros
    converted_amount = round(amount/exchange_rate)

    return converted_amount

currency = {'Botswana Pula(P)':'BWP',
            'Brazilian Real(R$)':'BRL',
            'Dollar($)':'USD',
            'Emirati Diram(AED)':'AED',
            'Indian Rupees(Rs.)':'INR',
            'Indonesian Rupiah(IDR)':'IDR',
            'NewZealand($)':'NZD',
            'Pounds(£)':'GBP',
            'Qatari Rial(QR)':'QAR',
            'Rand(R)':'ZAR',
            'Sri Lankan Rupee(LKR)':'LKR',
            'Turkish Lira(TL)':'TRY'}

df1['converted_avg_cost_for_two'] = df1[['currency', 'average_cost_for_two']].apply(lambda x: convert_currency(x['average_cost_for_two'], currency[x['currency']]), axis=1)
df1.tail()

restaurant_id                    restaurant_name country      city  \
7522        5912546                             Eataly  Turkey  İstanbul   
7523        5913006  Tarihi Çınaraltı Aile Çay Bahçesi  Turkey  İstanbul   
7524        5923535             Boon Cafe & Restaurant  Turkey  İstanbul   
7525        5914190                   Kanaat Lokantası  Turkey  İstanbul   
7526        5913508                            Katibim  Turkey  İstanbul   

                                                address  \
7522  Zorlu Center AVM, Köprü Katı, Levazım Mahalles...   
7523  Çengelköy Mahallesi, Çınaraltı Camii Sokak, No...   
7524  Çengelköy Mahallesi, Çengelköy Caddesi, Kara S...   
7525  Sultantepe Mahallesi, Selmani Pak Caddesi, No ...   
7526  Mimar Sinan Mahallesi, Harem Sahilyolu, No 53,...   

                                 locality  \
7522  Zorlu Center AVM, Levazım, Beşiktaş   
7523                     Çengelköy Merkez   
7524                     Çengelköy Merkez   
7525                       Üsküdar Merkez   
7526                       Üsküdar Merkez   

                                   locality_verbose  longitude  latitude  \
7522  Zorlu Center AVM, Levazım, Beşiktaş, İstanbul      29.02     41.07   
7523                     Çengelköy Merkez, İstanbul      29.05     41.05   
7524                     Çengelköy Merkez, İstanbul      29.05     41.05   
7525                       Üsküdar Merkez, İstanbul      29.02     41.03   
7526                       Üsküdar Merkez, İstanbul      29.01     41.03   

             cuisines  average_cost_for_two          currency  \
7522          Italian                   300  Turkish Lira(TL)   
7523        Fast Food                    45  Turkish Lira(TL)   
7524  Restaurant Cafe                   140  Turkish Lira(TL)   
7525        Home-made                    95  Turkish Lira(TL)   
7526  Restaurant Cafe                   150  Turkish Lira(TL)   

      has_table_booking  has_online_delivery  is_delivering_now price_range  \
7522                  0                    0                  0     gourmet   
7523                  0                    0                  0      normal   
7524                  0                    0                  0     gourmet   
7525                  0                    0                  0   expensive   
7526                  0                    0                  0     gourmet   

      aggregate_rating rating_color rating_text  votes color_nome  \
7522              4.30       5BA829   Very Good   1367      green   
7523              4.50       3F7E00   Excellent   1172  darkgreen   
7524              4.20       5BA829   Very Good   1160      green   
7525              4.00       5BA829   Very Good    770      green   
7526              4.20       5BA829   Very Good   1141      green   

      converted_avg_cost_for_two  
7522                           9  
7523                           1  
7524                           4  
7525                           3  
7526                           5

In [24]:
df1['average_cost_for_two'].describe()

count       6929.00
mean        7520.08
std       302534.78
min            0.00
25%           60.00
50%          250.00
75%          600.00
max     25000017.00
Name: average_cost_for_two, dtype: float64

In [25]:
df1['converted_avg_cost_for_two'].describe()

count       6929.00
mean        3637.53
std       300334.01
min            0.00
25%            7.00
50%           16.00
75%           40.00
max     25000017.00
Name: converted_avg_cost_for_two, dtype: float64

In [26]:
# Excluindo outlier na coluna average_cost_for_two

df1.drop(df1[df1['average_cost_for_two']==25000017].index, inplace=True)

# Data Analysis

In [27]:
df1.head()

restaurant_id             restaurant_name      country            city  \
0        6310675  Mama Lou's Italian Kitchen  Philippines  Las Piñas City   
2        6314542                   Blackbird  Philippines     Makati City   
3        6301293                    Banapple  Philippines     Makati City   
4        6315689                    Bad Bird  Philippines     Makati City   
5        6304833                       Manam  Philippines     Makati City   

                                             address  \
0  Block 1, Lot 36, Tropical Avenue Corner Tropic...   
2  Nielson Tower, Ayala Triangle Gardens, Salcedo...   
3  Ayala Triangle Gardens, Salcedo Village, Makat...   
4  Hole In The Wall, Floor 4, Century City Mall, ...   
5  Level 1, Greenbelt 2, Ayala Center, Greenbelt,...   

                                            locality  \
0                                   BF International   
2  Ayala Triangle Gardens, Salcedo Village, Makat...   
3  Ayala Triangle Gardens, Salcedo Village, Makat...   
4          Century City Mall, Poblacion, Makati City   
5              Greenbelt 2, San Lorenzo, Makati City   

                                    locality_verbose  longitude  latitude  \
0                   BF International, Las Piñas City     121.01     14.45   
2  Ayala Triangle Gardens, Salcedo Village, Makat...     121.02     14.56   
3  Ayala Triangle Gardens, Salcedo Village, Makat...     121.02     14.56   
4  Century City Mall, Poblacion, Makati City, Mak...     121.03     14.57   
5  Greenbelt 2, San Lorenzo, Makati City, Makati ...     121.02     14.55   

   cuisines  average_cost_for_two          currency  has_table_booking  \
0   Italian                  1100  Botswana Pula(P)                  1   
2  European                  3100  Botswana Pula(P)                  0   
3  Filipino                   800  Botswana Pula(P)                  0   
4  American                   700  Botswana Pula(P)                  0   
5  Filipino                   700  Botswana Pula(P)                  0   

   has_online_delivery  is_delivering_now price_range  aggregate_rating  \
0                    0                  0   expensive              4.60   
2                    0                  0     gourmet              4.70   
3                    0                  0   expensive              4.40   
4                    0                  0   expensive              4.40   
5                    0                  0   expensive              4.70   

  rating_color rating_text  votes color_nome  converted_avg_cost_for_two  
0       3F7E00   Excellent    619  darkgreen                          81  
2       3F7E00   Excellent    469  darkgreen                         228  
3       5BA829   Very Good    867      green                          59  
4       5BA829   Very Good    858      green                          52  
5       3F7E00   Excellent    930  darkgreen                          52

## Geral

In [28]:
rest_unicos = len(df1['restaurant_id'].unique())
print(f'Quantidade de restaurantes únicos cadastrados: {rest_unicos}')

Quantidade de restaurantes únicos cadastrados: 6928


In [29]:
paises_unicos = len(df1['country'].unique())
print(f'Quantidade de países únicos cadastrados: {paises_unicos}')

Quantidade de países únicos cadastrados: 15


In [30]:
cidades_unicas = len(df1['city'].unique())
print(f'Quantidade de cidades únicas cadastradas: {cidades_unicas}')

Quantidade de cidades únicas cadastradas: 125


In [31]:
qtd_avaliacoes = df1['votes'].sum()
print(f'Quantidade de avaliações feitas: {qtd_avaliacoes}')

Quantidade de avaliações feitas: 4194330


In [32]:
tipos_culinaria = len(df1['cuisines'].unique())
print(f'Quantidade de tipos de culinária registrados: {tipos_culinaria}')

Quantidade de tipos de culinária registrados: 165


## País

In [33]:
city_per_country = df1[['country', 'city']].groupby('country').nunique().sort_values(by='city', ascending=False).reset_index()
print(f'O país com o maior número de cidades registradas é a(o) {city_per_country.iloc[0,0]} ({city_per_country.iloc[0,1]} cidades)')

O país com o maior número de cidades registradas é a(o) India (49 cidades)


In [34]:
rest_per_country = df1[['country', 'restaurant_id']].groupby('country').nunique().sort_values(by='restaurant_id', ascending=False).reset_index()
print(f'O país com o maior número de restaurantes registrados é a(o) {rest_per_country.iloc[0,0]} ({rest_per_country.iloc[0,1]} restaurantes)')

O país com o maior número de restaurantes registrados é a(o) India (3111 restaurantes)


In [35]:
rest_4_per_country = df1[df1['price_range']=='gourmet'][['country', 'restaurant_id']].groupby('country').nunique().sort_values(by='restaurant_id', ascending=False).reset_index()
print(f'O país com o maior número de restaurantes com nível de preço 4 registrados é a(o) {rest_4_per_country.iloc[0,0]} ({rest_4_per_country.iloc[0,1]} restaurantes)')

O país com o maior número de restaurantes com nível de preço 4 registrados é a(o) United States of America (415 restaurantes)


In [36]:
cuisine_per_country = df1[['country', 'cuisines']].groupby('country').nunique().sort_values(by='cuisines', ascending=False).reset_index()
print(f'O país com o maior número de culinárias distantas registradas é a(o) {cuisine_per_country.iloc[0,0]} ({cuisine_per_country.iloc[0,1]} culinárias)')

O país com o maior número de culinárias distantas registradas é a(o) India (77 culinárias)


In [37]:
votes_per_country = df1[['country', 'votes']].groupby('country').sum().sort_values(by='votes', ascending=False).reset_index()
print(f'O país com a maior quantidade de avaliações é a(o) {votes_per_country.iloc[0,0]} ({votes_per_country.iloc[0,1]} avaliações)')

O país com a maior quantidade de avaliações é a(o) India (2800164 avaliações)


In [38]:
delivering_per_country = df1[df1['is_delivering_now']==1][['country', 'restaurant_id']].groupby('country').nunique().sort_values(by='restaurant_id', ascending=False).reset_index()
print(f'O país com o maior número de restaurantes que fazem entrega é a(o) {delivering_per_country.iloc[0,0]} ({delivering_per_country.iloc[0,1]} restaurantes)')

O país com o maior número de restaurantes que fazem entrega é a(o) India (1150 restaurantes)


In [39]:
booking_per_country = df1[df1['has_table_booking']==1][['country', 'restaurant_id']].groupby('country').nunique().sort_values(by='restaurant_id', ascending=False).reset_index()
print(f'O país com o maior número de restaurantes que aceitam reservas é a(o) {booking_per_country.iloc[0,0]} ({booking_per_country.iloc[0,1]} restaurantes)')

O país com o maior número de restaurantes que aceitam reservas é a(o) India (256 restaurantes)


In [40]:
votes_avg_per_country = df1[['country', 'votes']].groupby('country').mean().sort_values(by='votes', ascending=False).reset_index()
print(f'O país com a maior quantidade de avaliações na média é a(o) {votes_avg_per_country.iloc[0,0]} ({votes_avg_per_country.iloc[0,1]} avaliações)')

O país com a maior quantidade de avaliações na média é a(o) Indonesia (1112.825 avaliações)


In [41]:
avg_rating_per_country = df1[['country', 'aggregate_rating']].groupby('country').mean().sort_values(by='aggregate_rating', ascending=False).reset_index()
print(f'O país que possui, na média, a maior nota média é a(o) {avg_rating_per_country.iloc[0,0]} ({round(avg_rating_per_country.iloc[0,1],2)} de nota)')

O país que possui, na média, a maior nota média é a(o) Indonesia (4.6 de nota)


In [42]:
avg_rating_per_country = df1[['country', 'aggregate_rating']].groupby('country').mean().sort_values(by='aggregate_rating', ascending=True).reset_index()
print(f'O país que possui, na média, a menor nota média é a(o) {avg_rating_per_country.iloc[0,0]} ({round(avg_rating_per_country.iloc[0,1],2)} de nota)')

O país que possui, na média, a menor nota média é a(o) Brazil (3.32 de nota)


In [43]:
avg_cost_per_country = df1[['country', 'average_cost_for_two']].groupby('country').mean().sort_values(by='average_cost_for_two', ascending=True)
avg_cost_per_country

average_cost_for_two
country                                       
Canada                                   41.86
England                                  43.51
United States of America                 55.03
New Zeland                               62.15
Australia                                71.20
Turkey                                  128.58
Brazil                                  138.81
Singapure                               141.44
United Arab Emirates                    153.72
Qatar                                   174.00
South Africa                            339.23
India                                   704.40
Philippines                            1227.83
Sri Lanka                              2579.38
Indonesia                            303000.00

In [44]:
converted_avg_cost_per_country = df1[['country', 'converted_avg_cost_for_two']].groupby('country').mean().sort_values(by='converted_avg_cost_for_two', ascending=True)
converted_avg_cost_per_country

converted_avg_cost_for_two
country                                             
Turkey                                          4.00
Sri Lanka                                       8.44
India                                           8.46
South Africa                                   18.23
Indonesia                                      19.34
Brazil                                         27.77
New Zeland                                     38.42
Canada                                         41.86
United Arab Emirates                           41.89
Qatar                                          47.74
United States of America                       55.03
England                                        55.67
Australia                                      71.20
Philippines                                    90.44
Singapure                                     141.44

## Cidade

In [48]:
rest_per_city = df1[['city', 'restaurant_id']].groupby('city').nunique().sort_values(by='restaurant_id', ascending=False).reset_index()
print(f'A cidade com o maior número de restaurantes registrados é {rest_per_city.iloc[0,0]} ({rest_per_city.iloc[0,1]} restaurantes)')

A cidade com o maior número de restaurantes registrados é Abu Dhabi (80 restaurantes)


In [49]:
rest4_per_city = df1[df1['aggregate_rating']>4][['city', 'restaurant_id']].groupby('city').nunique().sort_values(by='restaurant_id', ascending=False).reset_index()
print(f'A cidade com o maior número de restaurantes com nota média acima de 4 é {rest4_per_city.iloc[0,0]} ({rest4_per_city.iloc[0,1]} restaurantes)')

A cidade com o maior número de restaurantes com nota média acima de 4 é Bangalore (79 restaurantes)


In [50]:
rest25_per_city = df1[df1['aggregate_rating']<2.5][['city', 'restaurant_id']].groupby('city').nunique().sort_values(by='restaurant_id', ascending=False).reset_index()
print(f'A cidade com o maior número de restaurantes com nota média abaixo de 2.5 é {rest25_per_city.iloc[0,0]} ({rest25_per_city.iloc[0,1]} restaurantes)')

A cidade com o maior número de restaurantes com nota média abaixo de 2.5 é Gangtok (33 restaurantes)


In [51]:
cost_per_city = df1[['city', 'average_cost_for_two']].groupby('city').mean().sort_values(by='average_cost_for_two', ascending=False).reset_index()
print(f'A cidade com o maior valor médio de um prato para dois é {cost_per_city.iloc[0,0]} ({round(cost_per_city.iloc[0,1],2)})')

A cidade com o maior valor médio de um prato para dois é Jakarta (309605.26)


In [52]:
converted_cost_per_city = df1[['city', 'converted_avg_cost_for_two']].groupby('city').mean().sort_values(by='converted_avg_cost_for_two', ascending=False).reset_index()
print(f'A cidade com o maior valor médio de um prato para dois (convertido para USD) é {converted_cost_per_city.iloc[0,0]} ({round(converted_cost_per_city.iloc[0,1],2)})')

A cidade com o maior valor médio de um prato para dois (convertido para USD) é Pasay City (294.5)


In [53]:
cuisines_per_city = df1[['city', 'cuisines']].groupby('city').nunique().sort_values(by='cuisines', ascending=False).reset_index()
print(f'A cidade com o maior número de culinárias distintas é {cuisines_per_city.iloc[0,0]} ({cuisines_per_city.iloc[0,1]} culinárias)')

A cidade com o maior número de culinárias distintas é Birmingham (32 culinárias)


In [54]:
booking_per_city = df1[df1['has_table_booking']==1][['city', 'restaurant_id']].groupby('city').nunique().sort_values(by='restaurant_id', ascending=False).reset_index()
print(f'A cidade com o maior número de restaurantes que fazem reserva é {booking_per_city.iloc[0,0]} ({booking_per_city.iloc[0,1]} restaurantes)')

A cidade com o maior número de restaurantes que fazem reserva é Bangalore (42 restaurantes)


In [55]:
delivery_per_city = df1[df1['is_delivering_now']==1][['city', 'restaurant_id']].groupby('city').nunique().sort_values(by='restaurant_id', ascending=False).reset_index()
print(f'A cidade com o maior número de restaurantes que fazem entrega é {delivery_per_city.iloc[0,0]} ({delivery_per_city.iloc[0,1]} restaurantes)')

A cidade com o maior número de restaurantes que fazem entrega é Vadodara (48 restaurantes)


In [56]:
online_per_city = df1[df1['has_online_delivery']==1][['city', 'restaurant_id']].groupby('city').nunique().sort_values(by='restaurant_id', ascending=False).reset_index()
print(f'A cidade com o maior número de restaurantes que fazem aceitam pedidos online é {delivery_per_city.iloc[0,0]} ({delivery_per_city.iloc[0,1]} restaurantes)')

A cidade com o maior número de restaurantes que fazem aceitam pedidos online é Vadodara (48 restaurantes)


## Restaurantes

In [57]:
votes_per_rest = df1[['restaurant_id', 'restaurant_name', 'votes']].sort_values(by=['votes', 'restaurant_id'], ascending=[False, True])
print(f'O restaurante com a maior quantidade de avaliações é a(o) {votes_per_rest.iloc[0,1]} ({votes_per_rest.iloc[0,2]} avaliações)')

O restaurante com a maior quantidade de avaliações é a(o) Bawarchi (41333 avaliações)


In [58]:
rating_per_rest = df1[['restaurant_id', 'restaurant_name', 'aggregate_rating']].sort_values(by=['aggregate_rating', 'restaurant_id'], ascending=[False,True])
print(f'O restaurante com a maior nota média é a(o) {rating_per_rest.iloc[0,1]} ({round(rating_per_rest.iloc[0,2],2)})')

O restaurante com a maior nota média é a(o) Indian Grill Room (4.9)


In [59]:
cost_per_rest = df1[['restaurant_id', 'restaurant_name', 'average_cost_for_two']].sort_values(by=['average_cost_for_two', 'restaurant_id'], ascending=[False, True])
print(f'O restaurante com o maior valor para duas pessoas é a(o) {cost_per_rest.iloc[0,1]} ({round(cost_per_rest.iloc[0,2],2)})')

O restaurante com o maior valor para duas pessoas é a(o) The Café - Hotel Mulia (1200000)


In [60]:
converted_cost_per_rest = df1[['restaurant_id', 'restaurant_name', 'converted_avg_cost_for_two']].sort_values(by=['converted_avg_cost_for_two', 'restaurant_id'], ascending=[False, True])
print(f'O restaurante com o maior valor para duas pessoas (convertido para USD) é a(o) {converted_cost_per_rest.iloc[0,1]} ({round(converted_cost_per_rest.iloc[0,2],2)})')

O restaurante com o maior valor para duas pessoas (convertido para USD) é a(o) Corner House (755)


In [61]:
lowest_rating_br = df1[df1['cuisines']=='Brazilian'][['restaurant_id', 'restaurant_name', 'aggregate_rating']].sort_values(by=['aggregate_rating','restaurant_id'], ascending=True)
print(f'O restaurante de culinária Brasileira com a pior nota média é a(o) {lowest_rating_br.iloc[0,1]} ({round(lowest_rating_br.iloc[0,2],2)})')

O restaurante de culinária Brasileira com a pior nota média é a(o) Loca Como tu Madre (0.0)


In [62]:
highest_rating_br = df1[(df1['cuisines']=='Brazilian')&(df1['country']=='Brazil')][['restaurant_id', 'restaurant_name', 'aggregate_rating']].sort_values(by=['aggregate_rating','restaurant_id'], ascending=[False, True])
print(f'O restaurante de culinária Brasileira e que é de Brasil com a maior nota média é a(o) {highest_rating_br.iloc[0,1]} ({round(highest_rating_br.iloc[0,2],2)})')

O restaurante de culinária Brasileira e que é de Brasil com a maior nota média é a(o) Braseiro da Gávea (4.9)


In [63]:
df1[['votes', 'has_online_delivery']].groupby('has_online_delivery').mean()

votes
has_online_delivery       
0                   479.48
1                   838.82

A média da quantidade de avaliações para os restaurante que aceitam pedidos online é consideravelmente maior do que a dos restaurantes que não aceitam pedido online.

In [64]:
df1[['has_table_booking', 'average_cost_for_two']].groupby('has_table_booking').mean()

average_cost_for_two
has_table_booking                      
0                               3488.60
1                              10499.57

In [65]:
df1[['has_table_booking', 'converted_avg_cost_for_two']].groupby('has_table_booking').mean()

converted_avg_cost_for_two
has_table_booking                            
0                                       29.10
1                                       35.90

A média do valor de um prato para duas pessoas é consideravelmente maior para restaurantes que aceitam reservas.

In [66]:
dfaux = df1.query("(country == 'United States of America') & (cuisines.isin(['BBQ', 'Japanese']))")
dfaux[['cuisines', 'average_cost_for_two']].groupby('cuisines').mean()

average_cost_for_two
cuisines                      
BBQ                      39.64
Japanese                 56.41

A média do valor de um prato para duas pessoas nos Estados Unidos é maior para restaurantes com a culinária Japonesa do que churrascarias (BBQ)

## Culinárias

In [67]:
np.sort(df1.cuisines.unique())

array(['Afghan', 'African', 'American', 'Andhra', 'Arabian', 'Argentine',
       'Armenian', 'Asian', 'Asian Fusion', 'Assamese', 'Australian',
       'Author', 'Awadhi', 'BBQ', 'Bakery', 'Balti', 'Bar Food',
       'Belgian', 'Bengali', 'Beverages', 'Biryani', 'Brazilian',
       'Breakfast', 'British', 'Burger', 'Burmese', 'Cafe', 'Cafe Food',
       'Cajun', 'California', 'Canadian', 'Cantonese', 'Caribbean',
       'Charcoal Chicken', 'Chettinad', 'Chinese', 'Coffee',
       'Coffee and Tea', 'Contemporary', 'Continental', 'Creole',
       'Crepes', 'Cuban', 'Deli', 'Desserts', 'Dim Sum', 'Dimsum',
       'Diner', 'Donuts', 'Drinks Only', 'Durban', 'Döner',
       'Eastern European', 'Egyptian', 'European', 'Fast Food',
       'Filipino', 'Finger Food', 'Fish and Chips', 'French',
       'Fresh Fish', 'Fusion', 'German', 'Giblets', 'Goan',
       'Gourmet Fast Food', 'Greek', 'Grill', 'Gujarati', 'Hawaiian',
       'Healthy Food', 'Home-made', 'Hyderabadi', 'Ice Cream', 'Indian',
 

In [68]:
italian_highest_rating = df1[df1['cuisines']=='Italian'][['restaurant_id', 'restaurant_name', 'aggregate_rating']].sort_values(by=['aggregate_rating','restaurant_id'], ascending=[False, True])
print(f'O restaurante de culinária Italiana com a maior nota média é a(o) {italian_highest_rating.iloc[0,1]} ({round(italian_highest_rating.iloc[0,2],2)})')

O restaurante de culinária Italiana com a maior nota média é a(o) Darshan (4.9)


In [69]:
dfaux = df1[df1['cuisines']=='Italian'][['restaurant_id', 'restaurant_name', 'aggregate_rating']].sort_values(by=['aggregate_rating','restaurant_id'], ascending=[True, True])
print(f'O restaurante de culinária Italiana com a menor nota média é a(o) {dfaux.iloc[0,1]} ({round(dfaux.iloc[0,2],2)})')

O restaurante de culinária Italiana com a menor nota média é a(o) Avenida Paulista (0.0)


In [70]:
dfaux = df1[df1['cuisines']=='American'][['restaurant_id', 'restaurant_name', 'aggregate_rating']].sort_values(by=['aggregate_rating','restaurant_id'], ascending=[False, True])
print(f'O restaurante de culinária Americana com a maior nota média é a(o) {dfaux.iloc[0,1]} ({round(dfaux.iloc[0,2],2)})')

O restaurante de culinária Americana com a maior nota média é a(o) Burger & Lobster (4.9)


In [71]:
dfaux = df1[df1['cuisines']=='American'][['restaurant_id', 'restaurant_name', 'aggregate_rating']].sort_values(by=['aggregate_rating','restaurant_id'], ascending=[True, True])
print(f'O restaurante de culinária Americana com a menor nota média é a(o) {dfaux.iloc[0,1]} ({round(dfaux.iloc[0,2],2)})')

O restaurante de culinária Americana com a menor nota média é a(o) Alston Bar & Beef (0.0)


In [72]:
dfaux = df1[df1['cuisines']=='Arabian'][['restaurant_id', 'restaurant_name', 'aggregate_rating']].sort_values(by=['aggregate_rating','restaurant_id'], ascending=[True, True])
print(f'O restaurante de culinária Árabe com a menor nota média é a(o) {dfaux.iloc[0,1]} ({round(dfaux.iloc[0,2],2)})')

O restaurante de culinária Árabe com a menor nota média é a(o) Raful (0.0)


In [73]:
dfaux = df1[df1['cuisines']=='Arabian'][['restaurant_id', 'restaurant_name', 'aggregate_rating']].sort_values(by=['aggregate_rating','restaurant_id'], ascending=[False, True])
print(f'O restaurante de culinária Árabe com a maior nota média é a(o) {dfaux.iloc[0,1]} ({round(dfaux.iloc[0,2],2)})')

O restaurante de culinária Árabe com a maior nota média é a(o) Mandi@36 (4.7)


In [74]:
dfaux = df1[df1['cuisines']=='Japanese'][['restaurant_id', 'restaurant_name', 'aggregate_rating']].sort_values(by=['aggregate_rating','restaurant_id'], ascending=[True, True])
print(f'O restaurante de culinária Japonesa com a menor nota média é a(o) {dfaux.iloc[0,1]} ({round(dfaux.iloc[0,2],2)})')

O restaurante de culinária Japonesa com a menor nota média é a(o) Banzai Sushi (0.0)


In [75]:
dfaux = df1[df1['cuisines']=='Japanese'][['restaurant_id', 'restaurant_name', 'aggregate_rating']].sort_values(by=['aggregate_rating','restaurant_id'], ascending=[False, True])
print(f'O restaurante de culinária Japonesa com a maior nota média é a(o) {dfaux.iloc[0,1]} ({round(dfaux.iloc[0,2],2)})')

O restaurante de culinária Japonesa com a maior nota média é a(o) Sushi Samba (4.9)


In [76]:
dfaux = df1[df1['cuisines']=='Home-made'][['restaurant_id', 'restaurant_name', 'aggregate_rating']].sort_values(by=['aggregate_rating','restaurant_id'], ascending=[True, True])
print(f'O restaurante de culinária Caseira com a menor nota média é a(o) {dfaux.iloc[0,1]} ({round(dfaux.iloc[0,2],2)})')

O restaurante de culinária Caseira com a menor nota média é a(o) GurMekan Restaurant (3.7)


In [77]:
dfaux = df1[df1['cuisines']=='Home-made'][['restaurant_id', 'restaurant_name', 'aggregate_rating']].sort_values(by=['aggregate_rating','restaurant_id'], ascending=[False, True])
print(f'O restaurante de culinária Caseira com a maior nota média é a(o) {dfaux.iloc[0,1]} ({round(dfaux.iloc[0,2],2)})')

O restaurante de culinária Caseira com a maior nota média é a(o) Kanaat Lokantası (4.0)


In [78]:
dfaux = df1[['cuisines', 'average_cost_for_two']].groupby('cuisines').mean().sort_values(by='average_cost_for_two', ascending=False).reset_index()
print(f'O tipo de culinária com o maior valor médio de um prato para duas pessoas é o(a) {dfaux.iloc[0,0]} ({round(dfaux.iloc[0,1],2)})')

O tipo de culinária com o maior valor médio de um prato para duas pessoas é o(a) Western (333333.33)


In [79]:
dfaux = df1[['cuisines', 'converted_avg_cost_for_two']].groupby('cuisines').mean().sort_values(by='converted_avg_cost_for_two', ascending=False).reset_index()
print(f'O tipo de culinária com o maior valor médio de um prato para duas pessoas (convertido para USD) é o(a) {dfaux.iloc[0,0]} ({round(dfaux.iloc[0,1],2)})')

O tipo de culinária com o maior valor médio de um prato para duas pessoas (convertido para USD) é o(a) New Mexican (180.0)


In [80]:
dfaux = df1[['cuisines', 'aggregate_rating']].groupby('cuisines').mean().sort_values(by='aggregate_rating', ascending=False).reset_index()
print(f'O tipo de culinária com a maior nota média é o(a) {dfaux.iloc[0,0]} ({round(dfaux.iloc[0,1],2)})')

O tipo de culinária com a maior nota média é o(a) Others (4.9)


In [81]:
dfaux = df1[(df1['has_online_delivery']==1)&(df1['is_delivering_now']==1)].groupby('cuisines').count()['restaurant_id'].reset_index()
dfaux = dfaux.sort_values(by='restaurant_id', ascending=False)
print(f'O tipo de culinária que possui mais restaurantes que aceitam pedido online e fazem entrega é o {dfaux.iloc[0,0]} ({dfaux.iloc[0,1]} restaurantes)')

O tipo de culinária que possui mais restaurantes que aceitam pedido online e fazem entrega é o North Indian (317 restaurantes)
